model/lifecycle.go

```diff
@@ -252,7 +252,7 @@ func SetTaskPriority(t task.Task, priority int64, caller string) error {
 
 	// negative priority - deactivate the task
 	if priority <= evergreen.DisabledTaskPriority {
-		if err = t.DeactivateTask(caller); err != nil {
+		if err = SetActiveState(caller, false, t); err != nil {
```

Comment:
There are a couple other key places where we currently skip the logic from `SetActiveState` when dealing with disabled tasks.  I think [SetDisabledPriority](https://github.com/evergreen-ci/evergreen/blob/main/model/task/task.go#L2200), [DisableTasks](https://github.com/evergreen-ci/evergreen/blob/main/model/task/task.go#L2227), [SetVersionPriority](https://github.com/evergreen-ci/evergreen/blob/main/model/lifecycle.go#L288), and [SetBuildPriority](https://github.com/evergreen-ci/evergreen/blob/main/model/lifecycle.go#L264) are instances that will need this change as well.

Reply:
Re our conversation `SetDisabledPriority` will be removed since it's functionality is covered by `DisableTasks` and `DisableTasks` will be moved up to the model package.